src/tests/ftest/osa/osa_dmg_negative_test.py

```diff
@@ -0,0 +1,152 @@
+
+        Args:
+            exp_result (str) : Expected result (Pass or Fail string).
+            dmg_output (str) : dmg output string.
+        """
+        if exp_result == "Pass":
+            self.is_rebuild_done(3)
+            if "succeeded" in dmg_output:
+                self.log.info("Test Passed")
+            else:
+                self.fail("Test Failed")
+        elif exp_result == "Fail":
+            if "failed" in dmg_output:
+                self.log.info("Test Passed")
+            else:
+                self.fail("Test Failed")
+        else:
+            self.fail("Invalid Parameter")
+
+    def run_osa_dmg_test(self, num_pool, extend=False):
+        """Run the offline extend without data.
+
+        Args:
+            num_pool (int) : total pools to create for testing purposes.
+            extend (bool) : Run testing after performing pool extend.
+        """
+        # Create a pool
+        self.dmg_command.exit_status_exception = False
+        pool = {}
+        pool_uuid = []
+
+        for val in range(0, num_pool):
+            pool[val] = TestPool(self.context, dmg_command=self.dmg_command)
+            pool[val].get_params(self)
+            # Split total SCM and NVME size for creating multiple pools.
+            pool[val].scm_size.value = int(pool[val].scm_size.value /
+                                           num_pool)
+            pool[val].nvme_size.value = int(pool[val].nvme_size.value /
+                                            num_pool)
+            pool[val].create()
+            pool_uuid.append(pool[val].uuid)
+            self.pool = pool[val]
+
+        # Start the additional servers and extend the pool
+        if extend is True:
+            self.log.info("Extra Servers = %s", self.extra_servers)
+            self.start_additional_servers(self.extra_servers)
+
+        # Get rank, target from the test_dmg_sequence
+        # Some test_dmg_sequence data will be invalid, valid.
+        for val in range(0, num_pool):
+            for i in range(len(self.test_seq)):
+                self.pool = pool[val]
+                if extend is True:
+                    scm_size = self.pool.scm_size
+                    nvme_size = self.pool.nvme_size
+                rank = self.test_seq[i][0]
+                target = "{}".format(self.test_seq[i][1])
+                expected_result = "{}".format(self.test_seq[i][2])
+                # Extend the pool
+                if extend is True:
+                    output = self.dmg_command.pool_extend(self.pool.uuid,
+                                                          rank,
+                                                          scm_size,
+                                                          nvme_size)
+                    self.log.info(output)
+                    self.validate_results(expected_result, output.stdout)
+                if (extend is False and (rank == "4" or rank == "5")):
```

Comment:
(pylint-consider-using-in) Consider merging these comparisons with "in" to "rank in ('4', '5')"